front/app/containers/Admin/formBuilder/components/FormFields/index.tsx

```diff
@@ -67,72 +63,167 @@ const FormFields = ({
+      return ` ${filteredPages.indexOf(field) + 1}`;
+    } else {
+      // Filter out pages
+      const filteredQuestion = formCustomFields.filter(
+        (customField) => customField.input_type !== 'page'
+      );
+      // Return question title
+      return ` ${filteredQuestion.indexOf(field) + 1}`;
+    }
+  };
+
   return (
     <DndProvider backend={HTML5Backend}>
-      <Box p="32px" height="100%" overflowY="auto">
+      <Box py="32px" height="100%" overflowY="auto">
         <List key={formCustomFields.length}>
           {formCustomFields.map((field, index) => {
             const hasErrors = !!errors.customFields?.[index];
             let outlineStyle = 'none';
             if (hasErrors) {
               outlineStyle = `1px solid ${colors.error}`;
-            } else if (selectedFieldId === field.id) {
-              outlineStyle = `1px solid ${colors.primary}`;
+            } else if (
+              isFieldSelected(field.id) &&
+              field.input_type !== 'page'
+            ) {
+              outlineStyle = `1px solid ${colors.teal300}`;
             }
             const fieldIdentifier = snakeCase(field.title_multiloc[locale]);
 
             return (
-              <Box
+              <FormFieldsContainer
+                role={'button'}
                 key={field.id}
-                style={{ outline: outlineStyle }}
+                style={{ outline: outlineStyle, outlineOffset: '-1px' }}
                 data-cy={`e2e-field-${fieldIdentifier}`}
+                background={getFieldBackgroundColor(field)}
+                onClick={() => {
+                  isEditingDisabled
+                    ? undefined
+                    : onEditField({ ...field, index });
+                }}
               >
                 <SortableRow
+                  iconFill={getTitleColor(field)}
+                  iconMargin="0px 0px 0px 24px"
                   id={field.id}
                   index={index}
                   moveRow={handleDragRow}
                   dropRow={() => {
                     // Do nothing, no need to handle dropping a row for now
                   }}
                 >
-                  <Box display="flex" className="expand">
-                    <Box as="span" display="flex" alignItems="center">
-                      <Text fontSize="base" my="0px" color="primary">
+                  <Box
+                    display="flex"
+                    justifyContent="space-between"
+                    className="expand"
+                  >
+                    <Box display="flex" alignItems="center">
+                      <Text
+                        as="span"
+                        color={getIndexTitleColor(field)}
+                        fontSize="base"
+                        mt="auto"
+                        mb="auto"
+                        fontWeight="bold"
+                        mx="12px"
+                      >
+                        {field.input_type === 'page' && (
```

Comment:
We can replace this block with something like this. The condition is on the messages

```
                        <>
                            <FormattedMessage {...(field.input_type === 'page' ? messages.page : messages.question)} />
                            {getIndexForTitle(formCustomFields, field)}
                        </>
```